### ISSUE TITLE
About test.py

### ISSUE THREAD

**Issue by BioDPJ** [microsoft/YOLaT-VectorGraphicsRecognition#3]
Dr.Jiang, 
Sorry to bother you. 
I run the command "CUDA_VISIBLE_DEVICES=0 python -u cad_recognition/test.py --data_dir data/FloorPlansGraph5_iter --pretrained_model log/run182_2_best.pth" with codes about "opt.arch" and "opt.graph" being commented out.
BUT before and then, I still got the errors:
"size mismatch for cls_net.fusion_block.0.weight: copying a param with shape torch.Size([1024, 128]) from checkpoint, the shape in current model is torch.Size([1024, 448]).
size mismatch for cls_net.fusion_block_super.0.weight: copying a param with shape torch.Size([1024, 128]) from checkpoint, the shape in current model is torch.Size([1024, 448]).
size mismatch for prediction_cls.0.0.weight: copying a param with shape torch.Size([512, 2304]) from checkpoint, the shape in current model is torch.Size([512, 2944])."
It really confusing since the model was saved based on "def save_checkpoint()" while it did not match during loading the model.
Would you like to resolve this issue? 
Thanks a lot and looking forward to your response soon.

Best regards, 
VivianBB.

**Comment by xinyangj** [comment]
Hi, this probably has something to do with the training architecture does not match with network arch in test.py (i.e. the --arch args). Please make sure they are the same.

**Comment by VivianBB** [comment]
thx for your time, Dr.Jiang. BUT I tried the train code just the same in the README.md. That's why confuses me. 
I wonder if there may be something wrong with the test code "CUDA_VISIBLE_DEVICES=0 python -u cad_recognition/test.py --data_dir data/FloorPlansGraph5_iter --pretrained_model log/run182_2_best.pth"?
If it possibly that some --args about testing phase I missed?

**Comment by VivianBB** [comment]
in case misleading, I tried set "--arch centernet3cc_rpn_gp_iter2  --graph bezier_cc_bb_iter" many times as stated in test.py

**Comment by BioDPJ** [comment]
Thx for your time, Dr.Jiang. I need to correct my comments just now.
I did try assign args like "--arch centernet3cc_rpn_gp_iter2  --graph bezier_cc_bb_iter". The problem is that the testing command could work out unless I add the args "--n_blocks_out 2", BUT the odd thing is the testing loss and loss_cls are over 684 (OMG, totally weird) with top acc over 90. It seems go for predicting the non-objects instead.
I have seen the loss curve in tensorboard, it seems well-functioned during training.
So, it actually has two issues with this command:
a. why do we have to add "--n_blocks_out" arg in test training, and b. why I tried different parameters, the testing losses are still so high?

 PS: The command about n_blocks in testing code :"CUDA_VISIBLE_DEVICES=0 python -u cad_recognition/test.py --data_dir data/FloorPlansGraph5_iter --phase test --n_blocks_out 2 --arch centernet3cc_rpn_gp_iter2  --graph bezier_cc_bb_iter --pretrained_model log/sem_seg_sparse-res-attr_edge-n2-C64-k16-drop0.0-lr0.00025_B4_20220314-202805_17e0fdb1-ddaf-4cfa-b79f-39a4dcb26998/checkpoint/run182_2_best.pth"

**Comment by lili2012** [comment]
@BioDPJ I meet the same problem. Did you solve it? How to solve it?
